Sources/SwiftKafka/KafkaProducerEvent.swift

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// An enumeration representing events that can be received through the ``KafkaProducerEvents`` asynchronous sequence.
+public enum KafkaProducerEvent: Hashable, Sendable {
+    /// A delivery report received from the Kafka cluster indicating the status of a produced message.
+    ///
+    /// Parameters:
+    ///    - results: Array of message acknowledgement results.
+    case deliveryReport(results: [Result<KafkaAcknowledgedMessage, KafkaAcknowledgedMessageError>])
```

Comment:
Plural and singular is mixed here
```suggestion
    /// Delivery reports received from the Kafka cluster indicating the status of a produced messages.
    case deliveryReports([Result<KafkaAcknowledgedMessage, KafkaAcknowledgedMessageError>])
```

With this we should probably introduce a new `KafkaDeliveryReport` that is basically a glorified `Result` just with a better name.